slam-libraries/viam-cartographer/src/slam_service/config.cc

```diff
@@ -30,6 +30,14 @@ DEFINE_int64(
     "Frequency at which we want to print map pictures while cartographer "
     "is running.");
 DEFINE_string(input_file_pattern, "", "Input file pattern");
+// TODO: Switch back to DEFINE_bool once PR #1689 is submitted
+// https://github.com/viamrobotics/rdk/pull/1689 This will allow intergation
+// tests to pass (See associated JIRA ticket:
+// https://viam.atlassian.net/browse/RSDK-1593)
+DEFINE_string(delete_processed_data, "",
```

Comment:
because a bool doesn't have a nil type like string and as John said in his avoid comment, we are capitalizing on the fact that RDK always will send a `delete_processed_data` value to ensure the integration tests for this and the RDK PR. 

After the other PR is merged we could keep this as a string to be consistent with orbslam but i think it makes more sense to move it to the bool form

Reply:
Update: I've pushed a revision that uses the bool flag as, now that delete_processed_data is required in SLAM (still optional in RDK) we can use the default false value of the bool to not delete data (which is what the current tests want).

Reply:
Updates to the orbslam TODOs have been made as well which should reduce the change needed in follow up PR